index/segment/mem/segment.go

```diff
@@ -82,6 +82,10 @@ func NewSegment(offset postings.ID, opts Options) (sgmt.MutableSegment, error) {
 	return s, nil
 }
 
+func (s *segment) Size() int64 {
+	return int64(int(s.readerID.Load()) - s.offset)
```

Comment:
nit: Might be cleaner to `return s.readerID.Load() - int64(s.offset)` to protect against precision (obviously not going to practically be a problem, so take it or leave it).

Reply:
Sure thing.

Reply:
done